CMakeLists.txt

```diff
@@ -199,7 +199,7 @@ endmacro()
 
 # Enabling all warnings in MSVC spams too much
 if(NOT MSVC)
-	add_definitions(-Wall)
+	add_definitions(-Wall -Wsign-conversion)
```

Comment:
if adding this does cause 6000 warnings, should it really be done in this pull request, or perhaps in a later one that also fixes a bunch of them, for example?